builtin/providers/azure/resource_azure_local_network.go

```diff
@@ -0,0 +1,270 @@
+// to create a virtual network on Azure.
+func resourceAzureLocalNetworkConnectionCreate(d *schema.ResourceData, meta interface{}) error {
+	azureClient, ok := meta.(*AzureClient)
+	if !ok {
+		return fmt.Errorf("Failed to convert to *AzureClient, got: %T", meta)
+	}
+	managementClient := azureClient.managementClient
+	networkClient := virtualnetwork.NewClient(managementClient)
+
+	log.Println("[INFO] Fetching current network configuration from Azure.")
+	azureClient.mutex.Lock()
+	defer azureClient.mutex.Unlock()
+	netConf, err := networkClient.GetVirtualNetworkConfiguration()
+	if err != nil {
+		return fmt.Errorf("Failed to get the current network configuration from Azure: %s", err)
+	}
+
+	// get provided configuration:
+	name := d.Get("name").(string)
+	vpnGateway := d.Get("vpn_gateway_address").(string)
+	var prefixes []string
+	if nprefixes := d.Get("address_space_prefixes.#").(int); nprefixes > 0 {
+		prefixes = []string{}
+		for i := 0; i < nprefixes; i++ {
+			prefixes = append(prefixes, d.Get(fmt.Sprintf("address_space_prefixes.%d", i)).(string))
+		}
+	}
+
+	// add configuration to network config:
+	netConf.Configuration.LocalNetworkSites = append(netConf.Configuration.LocalNetworkSites,
+		virtualnetwork.LocalNetworkSite{
+			Name:              name,
+			VPNGatewayAddress: vpnGateway,
+			AddressSpace: virtualnetwork.AddressSpace{
+				AddressPrefix: prefixes,
+			},
+		})
+
+	// send the configuration back to Azure:
+	log.Println("[INFO] Sending updated network configuration back to Azure.")
+	reqID, err := networkClient.SetVirtualNetworkConfiguration(netConf)
+	if err != nil {
+		return fmt.Errorf("Failed setting updated network configuration: %s", err)
+	}
+	err = managementClient.WaitForOperation(reqID, nil)
+	if err != nil {
+		return fmt.Errorf("Failed updating the network configuration: %s", err)
+	}
+
+	d.SetId(getRandomStringLabel(50))
+	return nil
+}
+
+// resourceAzureLocalNetworkConnectionRead does all the necessary API calls to
+// read the state of our local natwork from Azure.
+func resourceAzureLocalNetworkConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	azureClient, ok := meta.(*AzureClient)
+	if !ok {
+		return fmt.Errorf("Failed to convert to *AzureClient, got: %T", meta)
+	}
+	managementClient := azureClient.managementClient
+	networkClient := virtualnetwork.NewClient(managementClient)
+
+	log.Println("[INFO] Fetching current network configuration from Azure.")
+	netConf, err := networkClient.GetVirtualNetworkConfiguration()
+	if err != nil {
+		return fmt.Errorf("Failed to get the current network configuration from Azure: %s", err)
+	}
+
+	var found bool
+	name := d.Get("name").(string)
+
+	// browsing for our network config:
+	for _, lnet := range netConf.Configuration.LocalNetworkSites {
+		if lnet.Name == name {
+			found = true
+			d.Set("vpn_gateway_address", lnet.VPNGatewayAddress)
+			d.Set("address_space_prefixes", lnet.AddressSpace.AddressPrefix)
+		}
```

Comment:
@svanharmelen pointed this out as well.
It has already been fixed on my rebased branch.